src/parser.pl

```diff
@@ -19,7 +19,7 @@
 var_symbol(V,E0,E) --> "$", token(Cs), { atom_chars(N, Cs), ( memberchk(N-V0, E0) -> V=V0, E=E0 ; V=_, E=[N-V|E0] ) }.
 
 %Atoms are just tokens:
-atom_symbol(A) --> token(Cs), { atom_codes(A0,Cs), downcase_atom(A0,A) }.
+atom_symbol(A) --> token([C0|Rest]), { code_type(C0, to_upper(L0)), atom_codes(A, [L0|Rest]) }.
```

Comment:
Only first letter is made lowercase. This is a good tradeoff as that way Python interop is not restricted via module.ClassName for instance while when a user is writing MeTTa code the user does not need to care about upper or lowercase in naming.